ui/components/institutional/qr-code-modal/qr-code-modal.js

```diff
@@ -0,0 +1,254 @@
+
+  const handleClose = useCallback(async () => {
+    await dispatch(mmiActions.setConnectionRequest(null));
+    onClose();
+  }, [dispatch, mmiActions, onClose]);
+
+  const handleError = useCallback((message, e) => {
+    const errorMessage = `${message} Please try again or contact support if the problem persists.`;
+    console.error(message, e);
+    setError(errorMessage);
+    captureException(e);
+  }, []);
+
+  const generateKeyPair = useCallback(async () => {
+    try {
+      const keyPair = await window.crypto.subtle.generateKey(
+        {
+          name: 'RSA-OAEP',
+          modulusLength: 4096,
+          publicExponent: new Uint8Array([1, 0, 1]),
+          hash: { name: 'SHA-256' },
+        },
+        true,
+        ['encrypt', 'decrypt'],
+      );
+
+      const exportedPublicKey = await window.crypto.subtle.exportKey(
+        'spki',
+        keyPair.publicKey,
+      );
+
+      setPublicKey(
+        btoa(String.fromCharCode(...new Uint8Array(exportedPublicKey))),
+      );
+
+      setPrivateKey(keyPair.privateKey);
+    } catch (e) {
+      handleError('An error occurred while generating cryptographic keys.', e);
+    }
+  }, [handleError]);
+
+  const decrypt = useCallback(
+    async (_privateKey, encryptedData) => {
+      try {
+        const encryptedBuffer = Uint8Array.from(atob(encryptedData), (c) =>
+          c.charCodeAt(0),
+        );
+        const decrypted = await window.crypto.subtle.decrypt(
+          { name: 'RSA-OAEP' },
+          _privateKey,
+          encryptedBuffer,
+        );
+        const decryptedString = new TextDecoder().decode(decrypted);
+        return JSON.parse(decryptedString);
+      } catch (e) {
+        handleError('An error occurred while decrypting data.', e);
+        throw e;
+      }
+    },
+    [handleError],
+  );
+
+  useEffect(() => {
+    generateKeyPair();
+  }, [generateKeyPair]);
+
+  useEffect(() => {
+    const decryptAndProcessData = async () => {
+      if (connectionRequest && privateKey) {
+        try {
+          const decryptedPayload = await decrypt(
+            privateKey,
+            connectionRequest.payload,
+          );
+
+          setDecryptedMessage(decryptedPayload);
+        } catch (e) {
+          // Error handling is managed by the decrypt function
+        }
+      }
+    };
+
+    decryptAndProcessData();
+  }, [connectionRequest, privateKey, decrypt]);
+
+  useEffect(() => {
+    if (publicKey && channelId) {
+      const value = JSON.stringify({
+        publicKey,
+        actionDescription: 'mmi:connect-request/qr-code?encrypted',
+        channelId,
+        traceId,
+      });
+
+      setQrCodeValue(value);
+    }
+  }, [publicKey, channelId, traceId]);
+
+  useEffect(() => {
+    const updateConnectRequests = async () => {
+      try {
+        setQrConnectionRequest(decryptedMessage);
+        handleClose();
+      } catch (e) {
+        handleError('An error occurred while updating connection requests.', e);
```

Comment:
Can you also test this error handling as well?

Reply:
To test the error handling inside the catch block when updating connection requests in the component, I need to simulate a scenario where setQrConnectionRequest fails. I don't think this will benefit us. Wdyt?

Reply:
The reason behind this one was i saw you have tests covering all other error catching scenario, if it is covered somewhere else we can leave this as an exception 🙏🏻